moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -445,6 +436,13 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
 
     target_joint_velocities *= joint_limit_scale;
 
+    // Apply smoothing to the velocity if a smoother was provided.
+    if (smoother_)
+    {
+      smoother_->reset(current_state.velocities);
+      smoother_->doSmoothing(target_state.velocities);
+    }
+
     // Adjust joint position based on scaled down velocity
     target_joint_positions = current_joint_positions + (target_joint_velocities * servo_params_.publish_period);
```

Comment:
This calculation doesn't seem to use `target_state.velocities` from the previous line at all... is this right?

I guess what is the relationship between `target_state.velocities` and `target_joint_velocities`? Are they linked by reference (if so, that's dangerous and not readable), or have we been using the wrong variables for smoothing?

(this was the case with the positions anyway, so maybe a carry over bug from before unless I'm missing something)

Reply:
`target_joint_velocities` is simply an `Eigen:Map` of  `target_state.velocities`.
Used for efficient computation, defined [here](https://github.com/ros-planning/moveit2/blob/47d92ef973680aea1badd6cc6080598f040a89f4/moveit_ros/moveit_servo/src/servo.cpp#L401)

Reply:
Btw, the reason `target_state.velocities` and `current_state.velocities` is used with the smoothing API instead of the Eigen map is because the smoothing plugin only accepts `std::vector`, otherwise all of the computations could be performed with the Eigen map.